File: TIENDARENEGULPY/Cliente.py
class Cliente:
    def __init__(self, nombre, telefono):
        self.nombre = nombre
        self.telefono = telefono
        self.productos_comprados = []

    def get_nombre(self):
        return self.nombre

    def set_nombre(self, nombre):
        self.nombre = nombre

    def get_telefono(self):
        return self.telefono

    def set_telefono(self, telefono):
        self.telefono = telefono

    def get_productos_comprados(self):
        return self.productos_comprados

    def comprar(self, producto):
        self.productos_comprados.append(producto)
        print(self.nombre + " ha comprado:", producto.get_nombre())

File: TIENDARENEGUL/src/Cliente.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.Scanner;

// Clase Cliente
class Cliente {
    private String nombre;
    private String telefono;
    private List<Producto> productosComprados;

    // Constructor
    public Cliente(String nombre, String telefono) {
        this.nombre = nombre;
        this.telefono = telefono;
        this.productosComprados = new ArrayList<>();
    }

    // Getters y Setters
    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getTelefono() {
        return telefono;
    }

    public void setTelefono(String telefono) {
        this.telefono = telefono;
    }

    public List<Producto> getProductosComprados() {
        return productosComprados;
    }

    // Método para realizar una compra
    public void comprar(Producto producto) {
        productosComprados.add(producto);
        System.out.println(nombre + " ha comprado: " + producto.getNombre());
    }
}